plugins/providers/bigquery/config.go

```diff
@@ -173,7 +169,18 @@ func (c *Config) validatePermission(value interface{}, resourceType string, clie
 	}
 
 	if resourceType == ResourceTypeDataset {
-		if !utils.ContainsString([]string{DatasetRoleReader, DatasetRoleWriter, DatasetRoleOwner}, permision) {
+		roles, err := client.getGrantAbleRoleForDataset()
+		if err != nil {
+			return nil, ErrCannotFetchDatabasePermission
+		}
+		findMatchingRole := false
+		for _, role := range roles {
+			if string(role) == permision {
+				findMatchingRole = true
+				break
+			}
+		}
```

Comment:
can we use `strings.Contains` for this?

Reply:
any specific reason to use contains

Reply:
Please correct me if I am wrong, this one is only try to find whether the `permision` inside `roles` right?

Reply:
@mabdh no it's not, it's checking for exact match